### PR TITLE
Adding user-behavior-insights

### PR DESCRIPTION
**Pull request by jzonthemtn**
### Description
Adds the [user-behavior-insights](https://github.com/opensearch-project/user-behavior-insights) plugin to the build.

### Issues Resolved
- #4886 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gaiksaya** [comment]
Hi @jzonthemtn 
Can you please resolve the merge conflicts here so that we can proceed with the merge?
Thanks!

**Comment by jzonthemtn** [comment]
@gaiksaya Sorry for the delay - it's updated.

File: manifests/3.0.0/opensearch-3.0.0.yml [hunk 1]
```diff
@@ -238,3 +238,9 @@ components:
       - anomaly-detection
       - sql
       - ml-commons
+  - name: user-behavior-insights
```
Comment (peterzhuamazon) [inline manifests/3.0.0/opensearch-3.0.0.yml:241]: What is the depends_on entry for this plugin?

Comment (jzonthemtn) [reply manifests/3.0.0/opensearch-3.0.0.yml:241]: Are the `depends_on` other OpenSearch plugins?

**Review by gaiksaya — APPROVED** [review]

**Pull request merged**